nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -300,11 +307,21 @@ public void setLastModified(String lastModified) {
     }
 
     public void setProperties(Map<String, Object> properties) {
-        this.properties = properties;
+        if (properties == null) {
+            this.properties = new HashMap<>();
```

Comment:
Because the constructor does not allow `null`, see [Document.java#L100](https://github.com/nuxeo/nuxeo-java-client/pull/15/files/6bb32276d4b6ecec6ac08ccb36185c3956ec4b54#diff-b69b44919e11acf8f79c5af5dd171ba5L100). And I agree with that: as property setter method is available, the `Map<String, Object> properties` itself should not be null in any case.